plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java

```diff
@@ -1384,6 +1387,85 @@ public void testRowLevelDelete()
                 .hasStackTraceContaining("Delete without primary key or partition key is not supported");
     }
 
+    // test polymorphic table function
+
+    @Test
+    public void testNativeQuerySelectFromNation()
+    {
+        assertQuery(
+                "SELECT * FROM TABLE(cassandra.system.query(query => 'SELECT name FROM tpch.nation WHERE nationkey = 0 ALLOW FILTERING'))",
+                "VALUES 'ALGERIA'");
+        assertQuery(
+                "SELECT name FROM TABLE(cassandra.system.query(query => 'SELECT * FROM tpch.nation')) WHERE nationkey = 0",
+                "VALUES 'ALGERIA'");
+        assertThat(query("SELECT * FROM TABLE(cassandra.system.query(query => 'SELECT * FROM tpch.nation')) WHERE nationkey = 0"))
+                .isNotFullyPushedDown(FilterNode.class);
+    }
+
+    @Test
+    public void testNativeQuerySelectFromTestTable()
+    {
+        String tableName = "tpch.test_select" + randomNameSuffix();
+        onCassandra("CREATE TABLE " + tableName + "(col BIGINT PRIMARY KEY)");
+
+        onCassandra("INSERT INTO " + tableName + "(col) VALUES (1)");
+        assertQuery(
+                "SELECT * FROM TABLE(cassandra.system.query(query => 'SELECT * FROM " + tableName + "'))",
+                "VALUES 1");
+
+        onCassandra("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testNativeQueryCreateStatement()
+    {
+        String tableName = "test_create" + randomNameSuffix();
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(cassandra.system.query(query => 'CREATE TABLE tpch." + tableName + "(col INT PRIMARY KEY)'))"))
+                .hasMessage("Handle doesn't have columns info");
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+    }
+
+    @Test
+    public void testNativeQueryInsertStatementTableDoesNotExist()
+    {
+        String tableName = "test_insert" + randomNameSuffix();
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(cassandra.system.query(query => 'INSERT INTO tpch." + tableName + "(col) VALUES (1)'))"))
+                .hasMessageContaining("unconfigured table");
```

Comment:
What's the rationale for this test ?

Reply:
To verify failure from Cassandra, not connector side.